### PR TITLE
Adds a button to share draft link + better 404 page

### LINKED ISSUE #502: Add button for creators to share draft link in org workspace

### LINKED ISSUE #503: Add an image to 404 error page

### LINKED ISSUE #506: Remove hash from links

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #503 
Fixes #502 
Fixes #506 

## Test Plan

- [x] Test Responsiveness
   - [ ] Laptop (1200px)
   - [ ] Tablet (760px)
   - [ ] Phone (320px)
- [x] Cross-Browser Testing
   - [ ] Chrome
   - [ ] Firefox
   - [ ] Safari (use BrowserStack)
- [x] Local Language Support
- [x] Wrote tests
- [x] Tested locally
- [x] Self-review
- [x] Comments have been added appropriately
- [ ] ~Check for bundle size [here](https://bundlephobia.com/) if adding a package~
- [x] Added relevant details like Labels/Projects/Milestones etc.
- [ ] ~Images have `alt` attributes~
- [ ] ~Any `target="_blank"` links have `rel="noopener"`~
- [x] Only SVGs are used as images. If PNGs are used, their size has been optimised.
- [x] Tested on staging
- [ ] Tested on an actual physical phone
- [ ] Tested on production

### CODE REVIEW

File: nginx.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+server {
+    # https://stackoverflow.com/a/61753597/7870587
+    location / {
+        root   /usr/share/nginx/html;
+        index  index.html index.htm;
+        try_files $uri /index.html;
+    }
+}
```
Comment (deepansh96) [inline nginx.conf:8]: In the stackoverflow post, a lot of other things are also present in the nginx.conf file, what about those?
Why are they not important?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,6 +26,7 @@ jobs:
       - name: Run Test Cases
         env:
           VUE_APP_MIXPANEL_PROJECT_TOKEN: ${{ secrets.VUE_APP_MIXPANEL_PROJECT_TOKEN }}
+          VUE_APP_FRONTEND: ${{ secrets.VUE_APP_FRONTEND }}
```
Comment (deepansh96) [inline .github/workflows/ci.yml:29]: Why was this added?

File: src/App.vue [hunk 3]
```diff
@@ -165,6 +165,11 @@ export default {
     window.removeEventListener("beforeunload", this.onClose);
   },
   mounted() {
+    // remove hash from the url if it is present
+    if (location.hash) {
+      location.replace(location.hash.replace("#", ""));
+    }
```
Comment (deepansh96) [inline src/App.vue:171]: Is it possible to write a test for this?

File: src/assets/images/link.svg [hunk 4]
```diff
@@ -0,0 +1 @@
...2.9-14.6,35.4     c0.6,12.9,4.8,26.3,14.6,35.4C661.9,158.8,674,164.5,687.5,164.5L687.5,164.5z"></path></g></g><g><g><path d="M959.5,336.5c0-25,0-50.1,0-75.1c0-39.9,0-79.7,0-119.6c0-9.1,0-18.2,0-27.3c0-12.8-5.6-26.3-14.6-35.4     c-8.7-8.7-22.9-15.2-35.4-14.6c-12.9,0.6-26.3,4.8-35.4,14.6c-9,9.8-14.6,21.8-14.6,35.4c0,25,0,50.1,0,75.1     c0,39.9,0,79.7,0,119.6c0,9.1,0,18.2,0,27.3c0,12.8,5.6,26.3,14.6,35.4c8.7,8.7,22.9,15.2,35.4,14.6c12.9-0.6,26.3-4.8,35.4-14.6     C953.8,362.1,959.5,350,959.5,336.5L959.5,336.5z"></path></g></g><g><g><path d="M548.2,546.5c13.2-13.2,26.5-26.5,39.7-39.7c31.5-31.5,63.1-63.1,94.6-94.6     c38.1-38.1,76.2-76.2,114.4-114.4c33.1-33.1,66.1-66.1,99.2-99.2c16-16,32.3-31.9,48.1-48.1c0.2-0.2,0.4-0.4,0.7-0.7     c9.1-9.1,14.6-22.5,14.6-35.4c0-12.3-5.4-26.9-14.6-35.4c-9.5-8.7-22-15.2-35.4-14.6c-13.3,0.6-25.8,5.1-35.4,14.6     c-13.2,13.2-26.5,26.5-39.7,39.7c-31.5,31.5-63.1,63.1-94.6,94.6c-38.1,38.1-76.2,76.2-114.4,114.4     c-33.1,33.1-66.1,66.1-99.2,99.2c-16,16-32.3,31.9-48.1,48.1c-0.2,0.2-0.4,0.4-0.7,0.7c-9.1,9.1-14.6,22.5-14.6,35.4     c0,12.3,5.4,26.9,14.6,35.4c9.5,8.7,22,15.2,35.4,14.6C526.1,560.6,538.6,556.1,548.2,546.5L548.2,546.5z"></path></g></g></g></svg>
```
Comment (deepansh96) [inline src/assets/images/link.svg:1]: Remove `height` `width` and `fill`

File: src/assets/images/lock.svg [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<svg height='100px' width='100px'  fill="#ffffff" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" x="0px" y="0px" viewBox="0 0 100 100" style="enable-background:new 0 0 100 100;" xml:space="preserve"><style type="text/css">
```
Comment (deepansh96) [inline src/assets/images/lock.svg:1]: Remove `height` `width` and `fill`

File: src/assets/images/twitter.svg [hunk 6]
```diff
@@ -1 +1 @@
...
...ttp://www.serif.com/" xmlns:xlink="http://www.w3.org/1999/xlink"><g><path d="M448,512l-384,0c-35.328,0 -64,-28.672 -64,-64l0,-384c0,-35.328 28.672,-64 64,-64l384,0c35.328,0 64,28.672 64,64l0,384c0,35.328 -28.672,64 -64,64Z" id="Dark_Blue" style="fill:#1da1f2;fill-rule:nonzero;"/><path d="M196.608,386.048c120.704,0 186.752,-100.096 186.752,-186.752c0,-2.816 0,-5.632 -0.128,-8.448c12.8,-9.216 23.936,-20.864 32.768,-34.048c-11.776,5.248 -24.448,8.704 -37.76,10.368c13.568,-8.064 23.936,-20.992 28.928,-36.352c-12.672,7.552 -26.752,12.928 -41.728,15.872c-12.032,-12.8 -29.056,-20.736 -47.872,-20.736c-36.224,0 -65.664,29.44 -65.664,65.664c0,5.12 0.64,10.112 1.664,14.976c-54.528,-2.688 -102.912,-28.928 -135.296,-68.608c-5.632,9.728 -8.832,20.992 -8.832,33.024c0,22.784 11.648,42.88 29.184,54.656c-10.752,-0.384 -20.864,-3.328 -29.696,-8.192l0,0.896c0,31.744 22.656,58.368 52.608,64.384c-5.504,1.536 -11.264,2.304 -17.28,2.304c-4.224,0 -8.32,-0.384 -12.288,-1.152c8.32,26.112 32.64,45.056 61.312,45.568c-22.528,17.664 -50.816,28.16 -81.536,28.16c-5.248,0 -10.496,-0.256 -15.616,-0.896c28.928,18.432 63.488,29.312 100.48,29.312" id="Logo__x2014__FIXED" style="fill:#fff;fill-rule:nonzero;"/></g></svg>
```
Comment (deepansh96) [inline src/assets/images/twitter.svg:1]: Remove `height` `width` and `fill`

File: src/assets/images/whatsapp.svg [hunk 7]
```diff
@@ -1 +1 @@
...
...8.836c10.88,4.697 19.374,7.501 25.997,9.603c10.924,3.469 20.866,2.98 28.723,1.806c8.761,-1.309 26.98,-11.029 30.781,-21.677c3.799,-10.649 3.799,-19.777 2.659,-21.678c-1.139,-1.902 -4.179,-3.043 -8.74,-5.325m-83.207,113.573l-0.061,0c-27.22,-0.011 -53.917,-7.32 -77.207,-21.137l-5.539,-3.287l-57.413,15.056l15.325,-55.959l-3.608,-5.736c-15.184,-24.145 -23.203,-52.051 -23.192,-80.704c0.033,-83.611 68.083,-151.635 151.756,-151.635c40.517,0.016 78.603,15.811 107.243,44.474c28.64,28.663 44.404,66.764 44.389,107.283c-0.035,83.617 -68.083,151.645 -151.693,151.645m129.102,-280.709c-34.457,-34.486 -80.281,-53.487 -129.103,-53.507c-100.595,0 -182.468,81.841 -182.508,182.437c-0.013,32.156 8.39,63.546 24.361,91.212l-25.892,94.545l96.75,-25.37c26.657,14.535 56.67,22.194 87.216,22.207l0.075,0c100.586,0 182.465,-81.852 182.506,-182.448c0.019,-48.751 -18.946,-94.59 -53.405,-129.076" style="fill:#fff;"/></g><defs><linearGradient gradientTransform="matrix(0,-512,-512,0,256.001,512)" gradientUnits="userSpaceOnUse" id="_Linear1" x1="0" x2="1" y1="0" y2="0"><stop offset="0" style="stop-color:#25cf43;stop-opacity:1"/><stop offset="1" style="stop-color:#61fd7d;stop-opacity:1"/></linearGradient></defs></svg>
```
Comment (deepansh96) [inline src/assets/images/whatsapp.svg:1]: Remove `height` `width` and `fill`

File: src/assets/locales/hi.js [hunk 8]
```diff
@@ -342,7 +343,19 @@ export default {
     },
   },
   error: {
-    404: "वेब पेज मौजुद नहीं हैं ",
+    404: {
+      title: "वेब पेज मौजुद नहीं हैं",
+      description:
+        "आप जहाँ जाना चाह रहे हैं, हम उसे नहीं ढूंढ पाए। आप अपने होम पर वापस जाने की कोशिश कर सकते हैं।",
```
Comment (deepansh96) [inline src/assets/locales/hi.js:349]:
```suggestion
        "आप जहाँ जाना चाह रहे हैं, हम uss page ko नहीं ढूंढ पाए। आप अपने होम पर वापस जाने की कोशिश कर सकते हैं।",
```

File: src/pages/Editor.vue [hunk 9]
```diff
@@ -1156,6 +1186,15 @@ export default {
     ]),
     ...mapActions("generic", ["showSharePlioDialog", "showEmbedPlioDialog"]),
     ...Utilities,
+    copyPlioDraftLink() {
+      // copies the link to the clipboard
+      var success = Utilities.copyToClipboard(
+        this.getPlioDraftLink(this.plioId, this.org)
+      );
```
Comment (deepansh96) [inline src/pages/Editor.vue:1193]: Because you've spread(`...`) `Utilities` object into the methods object of this component, you can directly use those methods using `this.`

So this can be `this.copyToClipboard`
I'm mentioning this because in some components you've used utilities methods directly, using `this.`

Personally, I don't like to spread external js files into the methods object of a component and use it like `Utilities.SOME_FUNCTION` because that way it's super clear but we can use whatever but let's keep it consistent

File: src/pages/Editor.vue [hunk 10]
```diff
@@ -1156,6 +1186,15 @@ export default {
     ]),
     ...mapActions("generic", ["showSharePlioDialog", "showEmbedPlioDialog"]),
     ...Utilities,
+    copyPlioDraftLink() {
+      // copies the link to the clipboard
```
Comment (deepansh96) [inline src/pages/Editor.vue:1190]: comment on the top

File: src/pages/Editor.vue [hunk 11]
```diff
@@ -1156,6 +1186,15 @@ export default {
...
+      var success = Utilities.copyToClipboard(
```
Comment (deepansh96) [inline src/pages/Editor.vue:1191]:
```suggestion
      let success = Utilities.copyToClipboard(
```
No need to `var` here right?

**Review by deepansh96 — CHANGES_REQUESTED** [review]

**Comment by deepansh96** [comment]
Screenshot 2021-11-05 at 9 58 59 AM
There's still no way to go back from this - I either have to preview or publish.

File: src/App.vue (hunk 3 above)
Comment (dalmia) [reply src/App.vue:171]: Screenshot 2021-11-05 at 12 24 54 PM
I don't think so - as `location` is a property of the `window` object. For all our tests, we've had to define mock functions for `window`'s properties. So, we'll have to write the implementation of `location.hash`, `location.replace` and we can only test our implementation.

**Review by deepansh96 — APPROVED** [review]
